source/MaterialXGenShader/ShaderGraph.cpp

```diff
@@ -69,14 +69,105 @@ void ShaderGraph::addOutputSockets(const InterfaceElement& elem)
     }
 }
 
+void ShaderGraph::createConnectedNodes(ElementPtr downstreamElement,
```

Comment:
Extracted out this method since the "add connections" method was very long.

Reply:
Nitpick: the ElementPtr's could be passed as `const ElementPtr&`, to avoid the extra reference counter increment/decrement when passed by value.